MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -162,8 +162,8 @@ open class RouteLegProgress: NSObject {
         guard let index = leg.steps.index(of: step) else {
             return nil
         }
-        if index > 0 {
-            return leg.steps[index-1]
+        if index > 0 && index > stepIndex {
```

Comment:
The stated purpose of this method is to return the step before the given step. Nothing about this method says it has to be used for swiping between steps in the UI – what if we try to use it in the future to discover something about the previous step?

This check should be moved to code that’s specifically about swiping between steps.